velox/functions/prestosql/tests/TransformKeysTest.cpp

```diff
@@ -0,0 +1,241 @@
+  // lambda to use.
+  auto inputMap = makeMapVector<int64_t, int32_t>(
+      size,
+      [](auto row) { return row % 5; },
+      [](auto row) { return row % 7; },
+      [](auto row) { return row % 11; },
+      nullEvery(13));
+  auto condition =
+      makeFlatVector<bool>(size, [](auto row) { return row % 3 == 1; });
+  auto input = makeRowVector({condition, inputMap});
+  auto signature = rowType("x", BIGINT(), "unused", INTEGER());
+  registerLambda("plus5", signature, input->type(), "x + 5");
+  registerLambda("minus3", signature, input->type(), "x - 3");
+
+  auto result = evaluate<MapVector>(
+      "transform_keys(c1, if (c0, function('plus5'), function('minus3')))",
+      input);
+
+  // Make 2 expected vectors: one for rows where condition is true and another
+  // for rows where condition is false.
+  auto expectedPlus5 = makeMapVector<int64_t, int32_t>(
+      size,
+      [](auto row) { return row % 5; },
+      [](auto row) { return row % 7 + 5; },
+      [](auto row) { return row % 11; },
+      nullEvery(13));
+  auto expectedMinus3 = makeMapVector<int64_t, int32_t>(
+      size,
+      [](auto row) { return row % 5; },
+      [](auto row) { return row % 7 - 3; },
+      [](auto row) { return row % 11; },
+      nullEvery(13));
+  ASSERT_EQ(size, result->size());
+  for (auto i = 0; i < size; i++) {
+    if (i % 3 == 1) {
+      ASSERT_TRUE(expectedPlus5->equalValueAt(result.get(), i, i))
+          << "at " << i << ": " << expectedPlus5->toString(i) << " vs. "
+          << result->toString(i);
+
+    } else {
+      ASSERT_TRUE(expectedMinus3->equalValueAt(result.get(), i, i))
+          << "at " << i << ": " << expectedMinus3->toString(i) << " vs. "
+          << result->toString(i);
+    }
+  }
+}
+
+TEST_F(TransformKeysTest, dictionaryWithUniqueValues) {
+  vector_size_t size = 1'000;
+
+  auto indices = makeIndicesInReverse(size);
+  auto input = makeRowVector(
+      {makeFlatVector<int16_t>(size, [](auto /* row */) { return 5; }),
+       wrapInDictionary(
+           indices,
+           size,
+           makeMapVector<int64_t, int32_t>(
+               size,
+               [](auto row) { return row % 5; },
+               [](auto row) { return row % 7; },
+               [](auto row) { return row % 11; },
+               nullEvery(13)))});
+
+  registerLambda(
+      "plus5",
+      rowType("x", BIGINT(), "unused", INTEGER()),
+      input->type(),
+      "x + c0");
+
+  auto result =
+      evaluate<BaseVector>("transform_keys(c1, function('plus5'))", input);
+
+  auto expectedResult = wrapInDictionary(
+      indices,
+      size,
+      makeMapVector<int64_t, int32_t>(
+          size,
+          [](auto row) { return row % 5; },
+          [](auto row) { return row % 7 + 5; },
+          [](auto row) { return row % 11; },
+          nullEvery(13)));
+  assertEqualVectors(expectedResult, result);
+}
+
+TEST_F(TransformKeysTest, dictionaryWithDuplicates) {
```

Comment:
Just curious, what does this case test? (e.g., what's the difference between this and dictionaryWithUniqueValues for transform_keys?)

Reply:
Technically speaking this function doesn't distinguish between different dictionaries. I was adding these tests for completeness. Ideally, we would enhance Fuzzer to support lambda functions to make sure we get coverage of inputs with all sorts of encodings.